app/src/org/commcare/interfaces/AndroidSortableEntityAdapter.java

```diff
@@ -17,7 +17,8 @@
 public abstract class AndroidSortableEntityAdapter extends SortableEntityAdapter {
 
     public AndroidSortableEntityAdapter(List<Entity<TreeReference>> entityList, Detail detail, NodeEntityFactory factory) {
-        super(entityList, detail, factory instanceof AsyncNodeEntityFactory);
+        super(entityList, detail,
+                !(factory instanceof AsyncNodeEntityFactory) || ((AsyncNodeEntityFactory)factory).isBlockingAsyncMode());
```

Comment:
Would it be worth shifting the "isBlockingAsyncMode()" to the normal node entity factory and just having that return false?

Reply:
It's kind of contradicting to have a `isBlockingAsyncMode()` for a non async factory and I think will just pollute the Node Entity factory contract